Dockerfile

```diff
@@ -263,6 +243,34 @@ COPY setup/markdownlint/problem-matcher.json /etc/laminas-ci/problem-matcher/mar
 COPY setup/phpunit/problem-matcher.json /etc/laminas-ci/problem-matcher/phpunit.json
 
 
+# Setup external tools
+COPY composer.json \
+    composer.lock \
+    /tools/
+
+RUN cd /tools \
+    # Install `ext-bcmath` as it seems to be a requirement for `roave/backward-compatibility-check`
+    && apt install -y php-bcmath \
+    && composer install \
+        --classmap-authoritative
+
+# Set default PHP version based on the `composer.json` `config.platform.php` setting
+RUN export DEFAULT_PHP_VERSION=$(jq -r '.config.platform.php | sub("(?<minor>[0-9.]).99$"; "\(.minor)")' /tools/composer.json) \
+    # Cleanup composer files from external tools folder
+    && rm /tools/composer.* \
+    && update-alternatives --set php /usr/bin/php$DEFAULT_PHP_VERSION \
+    && update-alternatives --set phpize /usr/bin/phpize$DEFAULT_PHP_VERSION \
+    && update-alternatives --set php-config /usr/bin/php-config$DEFAULT_PHP_VERSION \
+    && update-alternatives --set phpdbg /usr/bin/phpdbg$DEFAULT_PHP_VERSION \
+    && echo "DEFAULT_PHP_VERSION=${DEFAULT_PHP_VERSION}" >> /etc/environment
+
+# Copy staabm/annotate-pull-request-from-checkstyle to external-tools stage
+RUN ln -s /tools/vendor/bin/cs2pr /usr/local/bin/cs2pr
+
+# Copy roave/backward-compatibility-check to this stage
+RUN ln -s /tools/vendor/bin/roave-backward-compatibility-check /usr/local/bin/roave-backward-compatibility-check
```

Comment:
Rather than symlinking these files, I'd expect them to be bash scripts running a fixed PHP version here.

Something like:

```sh
#!/usr/bin/env bash

# <unofficial bash strict mode here>

/path/to/exact/php/version/binary /tools/vendor/bin/roave-backward-compatibility-check $@
```

This way, there is no choosing the right PHP version for these tools: it's whatever we baked in this container upfront.

Reply:
That would work as well, but that would pass, for example, PHP 5.6 to the container and then runs PHP 8.2 for the roave script. I found that would be pretty weird and thus implemented `@default` .

But I can, if you insist, implement a bash script here, but that might also vary from the `@default` in case it is not maintained properly AND it will also diverge from what we use when running `composer install` during container build. Thats why I did not implemented something like this.

Reply:
Just as an FYI, if we go this route, to enable "strict mode" in bash: 

```bash
set -e
```

(which exits with a non-zero status if any command executed in the script exits with a non-zero status)

Reply:
The `roave/backward-compatibility-check` tooling will only ever work on latest PHP, so IMO we should completely ignore the PHP version requested, and treat this as a completely opaque binary that will run whatever it wants.

Yes, that will bring problems with extensions (will be unable to find some symbols), but consider that it will do a `composer install` on its own anyway, under the hood :D

Reply:
this is why I added `@default` as it is a php Version which will support the bc Tool... So I do not have to worry about it at all. This is the whole meaning of why its added. no1 needs to know what Version, no one has to send garbage php Version... So Whatever default php Version is installed is a Version which supports the Tool.

Reply:
Ok, works for me :+1: 

From my PoV, should probably have been omitted completely, but I'm fine with the current solution :D

Reply:
kk. Its all about transparency.
Having the following job while the command being executed with PHP 8.1 feels weird and thus I wanted to have something more clear.

```{"php": "7.4", "command": "roave-backward-compatibility"}``` 

Might not be the last tool and thus, it looked good enough for me. 😅